docs/users/cross-compilation.md

```diff
@@ -0,0 +1,144 @@
+---
+id: cross-compilation
+title: Cross compilation setups
+---
+
+Cross-building the same codebase with Scala 2.13 and Scala 3.3+ is common, but
+it collides with how Scalafix resolves rules and compiler options. A rule or
+`--scalac-options` flag that only works on Scala 2 can crash Scalafix when the
+Scala 3 compilation unit is processed, and vice versa. Until Scalafix grows
+conditional configuration, the safest way to avoid those failures is to split
+your configuration per Scala version and let the build tool wire the right file.
+
+## Why separate configs are required today
+
+* Scala 2 only flags such as `-Ywarn-unused-import` or SemanticDB options using
+  `-P:semanticdb` are rejected by Scala 3.
+* Some built-in or custom rules depend on compiler symbols that only exist on
+  one major version.
+* The CLI currently ingests a single `.scalafix.conf`; there is no per-target
+  override like sbt’s `CrossVersion`.
+
+## File layout suggestion
+
+Keep shared defaults in one file and let version-specific files `include` it via
+the HOCON `include` syntax. One convenient layout is:
+
+```
+.
+├── project/
+│   └── build config…
+└── scalafix/
+    ├── common.conf
+    ├── scala2.conf
+    └── scala3.conf
+```
+
+`common.conf`
+```scala
+rules = [
+  DisableSyntax,
+  OrganizeImports
+]
+
+DisableSyntax {
+  noFinalize = true
+}
+```
+
+`scala2.conf`
+```scala
+include "common.conf"
+
+rules += RemoveUnused
+
+// Scala 2 only compilers flags or rule settings
```

Comment:
As mentioned elsewhere, we should focus on compatibility (it's not necessarily about compiler flags but it can be new language features - although that's mostly for community rules, that you don't mention)
```suggestion
// Scala 2 only rules or rule settings
```